### ISSUE TITLE
core-ui not compatible with react-router-dom 5.0.0

### ISSUE THREAD

**Issue by MuhammadSulman** [ReactTraining/react-router#6720]
## Version

**Please replace this sentence with the React Router version that you are using.**

## Test Case

https://codesandbox.io/s/7zwjlm2j3q

**Comment by MuhammadSulman** [comment]
core ui working fine with  ```react-router-dom: "~4.3.1"```  but gives error when I try to use with ```react-router-dom": "5.0.0"```

its solution describe at https://github.com/ReactTraining/react-router/releases/tag/v4.4.0-beta.1 but still face same error 

```modules.js?hash=64eadce6a41dd5bd128fd48eb8bcb293f3d5a29f:119104 Uncaught Invariant Violation: You should not use <Route> or withRouter() outside a <Router>
    at invariant (http://localhost:3002/packages/modules.js?hash=64eadce6a41dd5bd128fd48eb8bcb293f3d5a29f:119104:15)
    at Route.computeMatch (http://localhost:3002/packages/modules.js?hash=64eadce6a41dd5bd128fd48eb8bcb293f3d5a29f:118231:29)
    at new Route (http://localhost:3002/packages/modules.js?hash=64eadce6a41dd5bd128fd48eb8bcb293f3d5a29f:118206:20)
    at constructClassInstance (http://localhost:3002/packages/modules.js?hash=64eadce6a41dd5bd128fd48eb8bcb293f3d5a29f:19692:18)
    at updateClassComponent (http://localhost:3002/packages/modules.js?hash=64eadce6a41dd5bd128fd48eb8bcb293f3d5a29f:23018:5)
    at beginWork (http://localhost:3002/packages/modules.js?hash=64eadce6a41dd5bd128fd48eb8bcb293f3d5a29f:23975:16)
    at performUnitOfWork (http://localhost:3002/packages/modules.js?hash=64eadce6a41dd5bd128fd48eb8bcb293f3d5a29f:27643:12)
    at workLoop (http://localhost:3002/packages/modules.js?hash=64eadce6a41dd5bd128fd48eb8bcb293f3d5a29f:27683:24)
    at HTMLUnknownElement.callCallback (http://localhost:3002/packages/modules.js?hash=64eadce6a41dd5bd128fd48eb8bcb293f3d5a29f:8526:14)
    at Object.invokeGuardedCallbackDev (http://localhost:3002/packages/modules.js?hash=64eadce6a41dd5bd128fd48eb8bcb293f3d5a29f:8576:16)
```


```
The above error occurred in the <Route> component:
    in Route (created by NavLink)
    in NavLink (created by AppSidebarNav)
    in li (created by NavItem)
    in NavItem (created by AppSidebarNav)
    in ul (created by Nav)
    in Nav (created by AppSidebarNav)
    in div (created by ScrollBar)
    in ScrollBar (created by AppSidebarNav)
    in AppSidebarNav (created by SideNav)
    in div (created by AppSidebar)
    in ClickOutComponent (created by AppSidebar)
    in AppSidebar (created by SideNav)
```

**Comment by MeiKatz** [comment]
This is not the fault of React Router but of Core UI. They should ensure that their library is not used with incompatible versions of libraries they depend on.

**Comment by StringEpsilon** [comment]
That's not a bug. The jump from 4.x to 5.x was made specifically because the two versions don't work together and people had problems like you describe here when it first released as 4.4.x. 

If you want to / have to use a library that uses 4.x, you have to simply use react router 4.x. 

The reason for the incomparability is the migration from the old react context API to the new context API.  RR 5.0 doesn't know about the context provided by RR 4.x components and vice versa. Hence you get the warning that components are used outside a Router. 

As far as I know, the only way to fix this would be to revert back to the old context API. That won't happen.

**Comment by MuhammadSulman** [comment]
@StringEpsilon you are right, this is what i have to do to use Core UI. Thanks !
